lesson2.html

```diff
@@ -0,0 +1,163 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+	<meta charset="UTF-8">
+	<title>Document</title>
+</head>
+<body>
+	<script>
+	/*1. Почему код дает именно такие результаты?
+
+		var a = 1, b = 1, c, d;
+		c = ++a; alert(c);           // 2
+		d = b++; alert(d);           // 1
+		c = (2+ ++a); alert(c);      // 5
+		d = (2+ b++); alert(d);      // 4
+		alert(a);                    // 3
+		alert(b);                    // 3
+
+		На второй строчке у значения переменной "а" префексная форм унарного оператора, поэтому  инкременирование
+		производится сразу что дает результат 2.
+
+		На третьей строчке у значения переменной "b" постпрефексная форма унарного оператора, поэтому сначала 
+		происходит возвращение значения, а потом выполняется инкременирование, поэтому вернулось заначение 1.
+		
+		На четвертой строчке c = (2+ ++a); //5 так как у "а" вновь префексная форм унарного оператора, 
+		"а" увеличивается снова  на единицу сразу, получилося а = 3; 
+
+		На пятой строчке d = (2+ b++);// 4  так как на третьей строчке у "b" уже была постпрефексная форма унарного оператора
+		то при повторном вызове "b" произошла операция инкременированирования и значение  "b" стало равнятся 2.
+
+		на шестой строчке alert(a); возращает значение 3 так как все опирации инкременирования закончены.
+
+		на седьмой срочке alert(b);  возращает значение 3 так как у "b" в предыдушем вызове опять же была постпрефексная форма унарного оператора, поэтом в этом вызове "b" вернуло число 3.
+
+	*/
+
+	/*2. Чему будет равен x?
+
+		var a = 2;
+		var x = 1 + (a *= 2);
+
+		x будет равен 5, (a *= 2) это сокращеная форма записи, можно записать иначе (a = a * 2) получается что "а" присвоили вырожение "а * 2"
+		
+  */
+
+  /*3. Объявить две целочисленные переменные — a и b и задать им произвольные начальные значения. Затем написать скрипт, который работает по следующему принципу:
+
+		a. если a и b положительные, вывести их разность;
+		b. если а и b отрицательные, вывести их произведение;
+		c. если а и b разных знаков, вывести их сумму;
+	*/
+
+  var a = 3;
+  var b = 12;
+  if(a >= 0 && b >= 0) {
+  	console.log(a - b);
+  }
+  else if(a < 0 && b < 0){
+  	console.log(a * b);
+	}
+  else if( a > 0 && b <= 0 || a <=0 && b > 0 ) {
```

Comment:
Здесь нет смысла в дополнительном условии. По сути ведь нам нужно покрыть "все остальные" случаи, поэтому хватит простого else
```suggestion
  else {
```